src/Compiler/Service/BackgroundCompiler.fsi

```diff
@@ -155,6 +155,10 @@ type internal IBackgroundCompiler =
         fileName: string * options: FSharpProjectOptions * sourceText: ISourceText option * userOpName: string ->
             (FSharpParseFileResults * FSharpCheckFileResults * SourceTextHash) option
 
+    abstract TryGetRecentCheckResultsForFile:
+        fileName: string * projectSnapshot: FSharpProjectSnapshot * userOpName: string ->
+            Async<(FSharpParseFileResults * FSharpCheckFileResults * SourceTextHash) option>
```

Comment:
Do we really need to return `SourceTextHash` from this API? Since we're guaranteeing to return a result for a given version, we could leave it up to the caller to ensure that's what they want. Then the API wouldn't have to be `async`.

It probably wouldn't have to be `async` anyway if we store the hash with the result. But not needing it at all would be better I think.

Reply:
Yeah, I'm very happy to remove that. It seemed pretty strange, anyway.